test/testsymboldatabase.cpp

```diff
@@ -2298,6 +2299,22 @@ class TestSymbolDatabase : public TestFixture {
         ASSERT(f && f->function() && !f->function()->isExtern() && f->function()->retDef->str() == "void");
     }
 
+    void externalFunctionsInsideAFunction() {
+        GET_SYMBOL_DB("void foo( void )\n"
+                      "{\n"
+                      "    extern void bar( void );\n"
+                      "    bar();\n"
+                      "}\n");
+
+        ASSERT(db && errout.str().empty());
+
+        const Token *f = Token::findsimplematch(tokenizer.tokens(), "bar");
+        ASSERT(f && f->function() && f->function()->isExtern() && f == f->function()->tokenDef && f->function()->retDef->str() == "void");
+
+        const Token *call = Token::findsimplematch(f->next(), "bar");
+        ASSERT(call && call->function() && call->function()->isExtern() && call->function() == f->function());
```

Comment:
hmm I feel the `call->function() && call->function()->isExtern() &&` is redundant. because you check that `call->function() == f->function()` and you already determined above that it's non-null and extern..